cache/common.go

```diff
@@ -0,0 +1,28 @@
+package cache
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"io"
+	"os"
+)
+
+// We need this prefix because there could be multiple restore steps in one workflow with multiple cache keys
+const cacheHitEnvVarPrefix = "BITRISE_CACHE_HIT__"
+
+func checksumOfFile(path string) (string, error) {
```

Comment:
You have two `checksumOfFile` named function in two separate packages which look exactly the same except one returns a string and the other a byte array. You could have a byte array one as the main one which holds the core of the logic and then the one which needs to return a string just wraps the byte array one and transforms its output and returns it.

Reply:
I'm not sure how I feel about sharing this function across packages. I would have to make the function public to make it accessible for both the outer and inner package 😔 But this is clearly not part of the public API, so I think in this case it's still better to have two slightly identical functions in the two packages.